core/templates/dev/head/pages/dashboard/Dashboard.js

```diff
@@ -205,7 +225,11 @@ oppia.controller('Dashboard', [
         } else {
           $scope.activeTab = 'myExplorations';
         }
-        $rootScope.loadingMessage = '';
+        if ($location.path() === '/back') {
```

Comment:
The reason for setting it here is, according to the previous discussion with @Arunabh98 whenever the back button is pressed the url is first redirected to '/back' and for normal functionality it must be redirected to the '/' and while that redirection occurs the Loading message is to be displayed instead of the previous instance of dashboard.
But I've figured out a work around that eliminates the use of '/back' entirely to prevent the redirection loop. I'll make a PR with the changes.